logbook/elisa.py

```diff
@@ -12,63 +13,68 @@ class ElisaLogbook:
     '''
     This class collects data relating the logbook into one object, and uses it to connect to ELisA.
     '''
-    def __init__(self, console, configuration):
-        self.console = console  #This seems useless, so foo is passed to it in all instances
+    def __init__(self, configuration, handler):
         self.elisa_arguments = {"connection": configuration['connection']}
         self.website = configuration['website']
         self.message_attributes = configuration['attributes']
         self.log = logging.getLogger(self.__class__.__name__)
         self.log.info(f'ELisA logbook connection: {configuration["website"]} (API: {configuration["connection"]})')
-        self.current_id = None
-        self.current_run = None
-        self.current_run_type = None
+        self.session_handler = handler
 
-    def _start_new_message_thread(self):
-        self.log.info("ELisA logbook: Next message will be a new thread")
-        self.current_id = None
-        self.current_run = None
-        self.current_run_type = None
+    def start_new_thread(self, subject:str, body:str, command:str, author:str, systems:[str]):
+        elisa_arg = copy.deepcopy(self.elisa_arguments)
+        elisa_user = credentials.get_login('elisa')
 
-    def _send_message(self, subject:str, body:str, command:str, author:str):
-            if not (credentials.check_kerberos_credentials()):
-                credentials.new_kerberos_ticket
+        with tempfile.NamedTemporaryFile() as tf:
+            try:
+                sso = {"ssocookie": self.session_handler.generate_elisa_cern_cookie(self.website, tf.name)}
+                elisa_arg.update(sso)
+                elisa_inst = Elisa(**elisa_arg)
+                answer = None
+                self.log.info("ELisA logbook: Creating a new message thread")
+                message = MessageInsert()
+                message.author = author
+                message.subject = subject
+                for attr_name, attr_data in self.message_attributes[command].items():
+                    if attr_data['set_on_new_thread']:
+                        setattr(message, attr_name, attr_data['value'])
```

Comment:
What is this doing?

Reply:
It searches through the config for attributes that need to be changed in the "message" object if we're starting a new thread, and changes them. For example in the np02_coldbox config, when it recieved the command start, and a new thread is being started, it will set message.type to "Automatic", and message.Automatic_Message_Type to "SoR".

Reply:
Ah this is the same as in nanorc. Should have remembered it.